src/converters/table-cell-refresh-post-fixer.js

```diff
@@ -25,17 +25,61 @@ export default function injectTableCellRefreshPostFixer( model ) {
 function tableCellRefreshPostFixer( model ) {
 	const differ = model.document.differ;
 
-	let fixed = false;
+	// Stores cells to be refreshed so the table cell will be refreshed once for multiple changes.
+	const cellToRefresh = new Set();
 
 	for ( const change of differ.getChanges() ) {
 		const parent = change.type == 'insert' || change.type == 'remove' ? change.position.parent : change.range.start.parent;
 
-		if ( parent.is( 'tableCell' ) ) {
-			differ.refreshItem( parent );
+		if ( parent.is( 'tableCell' ) && checkRefresh( parent, change.type ) ) {
+			cellToRefresh.add( parent );
+		}
+	}
 
-			fixed = true;
+	if ( cellToRefresh.size ) {
+		for ( const tableCell of cellToRefresh.values() ) {
+			differ.refreshItem( tableCell );
 		}
+
+		return true;
+	}
+
+	return false;
+}
+
+// Check if the model table cell requires refreshing to be re-rendered to a proper state in the view.
+//
+// This methods detects changes that will require renaming <span> to <p> (or vice versa) in the view,
+//
+// This method is simple heuristic that check only single change and will give false positive result when many changes will result
+// in a state that does not require renaming in the view.
+//
+// For instance: The <span> to <p> should be renamed when adding an attribute to a <paragraph>.
+// But adding one attribute and removing another will result in false positive: the check for added attribute will see one attribute
+// on a paragraph and will falsy qualify such change as adding attribute to a paragraph without any attribute.
+//
+// @param {module:engine/model/element~Element} tableCell Table cell to check.
+// @param {String} type Type of change.
+function checkRefresh( tableCell, type ) {
+	// If all children of a table cell were removed - refresh it.
+	if ( !tableCell.childCount ) {
```

Comment:
Actually, now I wonder, do we have to do that? Didn't we have another post-fixer for that earlier? 🤔

Reply:
What I mean is that we need to auto-paragraph this cell rather than refresh it, but refreshing it might trigger auto-paragraphing (in other post-fixer). And I just wonder if this shouldn't be elsewhere (if it is needed) or removed.

Reply:
yeah.. probably also

Reply:
fixing

Reply:
fixed

Reply:
fixed